// @ngneat/overview/esm2022/lib/dynamic-view/dynamic-view.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class DynamicViewComponent {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.4", ngImport: i0, type: DynamicViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.0.4", type: DynamicViewComponent, isStandalone: true, selector: "dynamic-view", inputs: { content: "content" }, ngImport: i0, template: ` <div [innerHTML]="content"></div> `, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.4", ngImport: i0, type: DynamicViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'dynamic-view',
                    standalone: true,
                    template: ` <div [innerHTML]="content"></div> `,
                }]
        }], propDecorators: { content: [{
                type: Input
            }] } });
