// @ngneat/overview/esm2022/lib/dynamic-view/dynamic-view.directive.mjs
import { Directive, Input, TemplateRef, ViewContainerRef, inject, } from '@angular/core';
import { isComponent, isString } from '../views/types';
import { ViewService } from '../views/view';
import { DynamicViewComponent } from './dynamic-view.component';
import * as i0 from "@angular/core";
export class DynamicViewDirective {
    constructor() {
        this.defaultTpl = inject(TemplateRef);
        this.vcr = inject(ViewContainerRef);
        this.viewService = inject(ViewService);
    }
    ngOnInit() {
        this.resolveContentType();
    }
    ngOnChanges(changes) {
        const viewChanged = changes.view && !changes.view.isFirstChange();
        const contextChanged = changes.context && !changes.context.isFirstChange();
        const inputsChanged = changes.inputs && !changes.inputs.isFirstChange();
        if (viewChanged) {
            this.resolveContentType();
        }
        else if (contextChanged) {
            this.viewRef.updateContext(this.context);
        }
        else if (isComponent(this.view) && inputsChanged) {
            this.viewRef.setInputs(this.inputs || {});
        }
    }
    resolveContentType() {
        this.viewRef?.destroy();
        if (isString(this.view)) {
            this.viewRef = this.viewService.createComponent(DynamicViewComponent, {
                vcr: this.vcr,
                injector: this.injector,
            });
            this.viewRef.setInput('content', this.view).detectChanges();
        }
        else if (isComponent(this.view)) {
            this.viewRef = this.viewService.createComponent(this.view, {
                vcr: this.vcr,
                injector: this.injector ?? this.vcr.injector,
                context: this.context,
            });
            if (this.inputs) {
                this.viewRef.setInputs(this.inputs);
            }
        }
        else {
            this.viewRef = this.viewService.createView(this.view || this.defaultTpl, {
                vcr: this.vcr,
                injector: this.injector ?? this.vcr.injector,
                context: this.context,
            });
        }
    }
    ngOnDestroy() {
        this.viewRef?.destroy();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.0.4", ngImport: i0, type: DynamicViewDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.0.4", type: DynamicViewDirective, isStandalone: true, selector: "[dynamicView]", inputs: { view: ["dynamicView", "view"], injector: ["dynamicViewInjector", "injector"], context: ["dynamicViewContext", "context"], inputs: ["dynamicViewInputs", "inputs"] }, usesOnChanges: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.0.4", ngImport: i0, type: DynamicViewDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[dynamicView]',
                    standalone: true,
                }]
        }], propDecorators: { view: [{
                type: Input,
                args: ['dynamicView']
            }], injector: [{
                type: Input,
                args: ['dynamicViewInjector']
            }], context: [{
                type: Input,
                args: ['dynamicViewContext']
            }], inputs: [{
                type: Input,
                args: ['dynamicViewInputs']
            }] } });
